omod/src/main/java/org/openmrs/module/fhir2/web/filter/ForwardingFilter.java

```diff
@@ -39,45 +40,26 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
 			
 			String contextPath = ((HttpServletRequest) req).getContextPath();
 			String prefix = contextPath + "/ws/fhir2";
-			if (requestURI.startsWith(prefix)) {
-				int prefixIdx = requestURI.indexOf(prefix);
-				if (prefixIdx >= 0) {
-					int prefixEnd = prefixIdx + prefix.length();
-					int pathIdx = requestURI.indexOf('/', prefixEnd + 1);
-					
-					if (pathIdx >= 0) {
-						String version = requestURI.substring(prefixEnd, pathIdx);
-						
-						if (version.isEmpty()) {
-							((HttpServletResponse) res).sendError(HttpServletResponse.SC_NOT_FOUND);
-							return;
-						}
-						
-						if (version.charAt(0) == '/') {
-							version = version.substring(1);
-						}
-						
-						String replacement;
-						switch (version) {
-							case "R3":
-								prefix += "/" + version;
-								replacement = "/ms/fhir2R3Servlet";
-								break;
-							case "R4":
-								prefix += "/" + version;
-								replacement = "/ms/fhir2Servlet";
-								break;
-							default:
-								((HttpServletResponse) res).sendError(HttpServletResponse.SC_NOT_FOUND);
-								return;
-						}
-						
-						String newURI = requestURI.replace(prefix, replacement);
-						req.getRequestDispatcher(newURI).forward(req, res);
-						return;
-					}
-				}
+			String fhirVersion = String.valueOf(FhirVersionUtils.getFhirResourceVersion(request));
+			
+			String replacement;
+			switch (fhirVersion) {
```

Comment:
It would be better to switch on the enum value here, even if we do need to calculate the string value. That way we have a compile-time check of the various cases.